src/components/DropList.tsx

```diff
@@ -0,0 +1,83 @@
+import { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[],
+  delay: number,
+  onSelected: (people: Person) => void,
+};
+
+export const DropList: React.FC<Props> = ({
+  people,
+  delay,
+  onSelected,
+}) => {
+  const [focus, setFocus] = useState(false);
+  const [queryPeople, setQueryPeople] = useState('');
+  const [appliedQueryPeople, setAppliedQueryPeople] = useState('');
+
+  const handleSelectPerson = (person: Person) => {
+    setQueryPeople(person.name);
+    onSelected(person);
+  };
+
+  const applyQueryPeople = useCallback(debounce(
+    setAppliedQueryPeople, delay,
+  ), []);
+
+  const handleQueryPeople = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQueryPeople(e.target.value);
+    applyQueryPeople(e.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(
+      (peop) => peop.name.toLowerCase().trim()
+        .includes(queryPeople.toLowerCase().trim()),
+    );
+  }, [appliedQueryPeople, people]);
+
+  return (
+    <div className={focus ? 'dropdown is-active' : 'dropdown'}>
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={queryPeople}
+          onChange={handleQueryPeople}
+          onFocus={() => setFocus(true)}
+          onBlur={() => setFocus(false)}
+        />
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        <div className="dropdown-content">
+          {
+            filteredPeople.length ? (
```

Comment:
convert to boolean in such cases for safety
```suggestion
            !!filteredPeople.length ? (
```